boards/shared/Src/App/App_FaultManager.c

```diff
@@ -0,0 +1,81 @@
+#include "App_CanRx.h"
+#include "App_CanTx.h"
+#include "App_FaultManager.h"
+
+#define _add_critical_fault(PREFIX, FAULT)       \
+    if (PREFIX##_GetPeriodicSignal_##FAULT(can)) \
+        return true;
+
+#define add_critical_fault(PREFIX, FAULT) _add_critical_fault(PREFIX, FAULT)
+
+static void *_fsm_can;
+static void *_dcm_can;
+static void *_pdm_can;
+static void *_dim_can;
+static void *_bms_can;
+
+void App_FaultManager_Init(
+    void *fsm_can,
+    void *dcm_can,
+    void *pdm_can,
+    void *dim_can,
+    void *bms_can)
+{
+    _fsm_can = fsm_can;
+    _dcm_can = dcm_can;
+    _pdm_can = pdm_can;
+    _dim_can = dim_can;
+    _bms_can = bms_can;
+}
+
+bool App_FaultManager_HasFsmCriticalFaultManager(void)
+{
+    void *can = _fsm_can;
```

Comment:
Could add `_fsm_can` as another argument to add_critical_fault (which means we could remove `void* can`) but I decided against it (for now) because it makes the macro more clunky to use